### PR TITLE
Update README.md specifying prerequisites.

### LINKED ISSUE #31: Update README file
The README file is not much descriptive and leaves certain aspects under assumption. 

Quoting @pri22296's comment https://github.com/fossasia/yaydoc/pull/30#issuecomment-302753608

>  Basically we need a directory with all markup files and an index.rst which contains toctrees to link the various documents. Basically what sphinx requires. could you add it to the README??

### PR DESCRIPTION
**Pull request by imujjwal96**
Fixes #31

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,27 +3,32 @@
...
+## Usage
+- Generate `Github Personal Token` by following this [Guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+- Set the following Environment Variables in Travis CI. [Guide](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings)     
+
+| Environment Variable | Description                                       | Default / FORMAT  |
+|:--------------------:| ------------------------------------------------- |:-----------------:|
+| AUTHOR               | Author of the repository.                         | NONE * |
+| DOCPATH              | Path of the documentation.                        | NONE * (eg. `docs/`) |
+| DOCTHEME             | Name of the theme.                                | alabaster ([built-in themes](http://www.sphinx-doc.org/en/stable/theming.html#builtin-themes)) | 
+| DOCURL               | Custom URL at which the site should be published. | <i>\<username or organization>.github.io/\<reponame></i> * |
```
Comment (pri22296) [inline README.md]: DOCURL is optional. By default github pages already pushes to `<username or organization>.gihub.io/<reponame>`. 
If only the docs should be published to a custom url, DOCURL is needed. see [this](https://help.github.com/articles/using-a-custom-domain-with-github-pages/) for reference.

Comment (sch00lb0y) [inline README.md]: add one more column says that required yes or no

**Comment by mariobehling** [comment]
Please resolve conflicts as well.

**Comment by imujjwal96** [comment]
Made the necessary changes while resolving conflicts.
Resolved conflicts via Github

File: README.md [hunk 2]
```diff
@@ -3,43 +3,46 @@
...
-  https://<username>:<token>@github.com/<organisation or username>/<repname>
-  ```
-- add the following to travis.yml
-
-If language is python
 
+### Prerequisites
+- Create a directory in your repository containing all the markup files along with an `index.rst` file which contains [toctrees](http://www.sphinx-doc.org/en/stable/markup/toctree.html) to link the various documents.
+- Create a `gh-pages` branch in your repository.
+- Register your repository with [Travis CI](https://www.google.co.in/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwjn8Y3v1_zTAhUKtI8KHaJWBssQFgg6MAA&url=https%3A%2F%2Ftravis-ci.org%2F&usg=AFQjCNFI_4340tr1MDD78A2HC-luYCxgAQ&sig2=d-UEeLAfCHrO-7wo9qYSGg).
```
Comment (pri22296) [inline README.md]: Do we need such a long URL? just use https://travis-ci.org/

File: README.md [hunk 3]
```diff
@@ -3,43 +3,46 @@
...
+
+## Usage
+- Generate `Github Personal Token` by following this [Guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+- Set the following Environment Variables in Travis CI. [Guide](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings)     
+
+| Environment Variable | Description                                       | Default / FORMAT  |
+|:--------------------:| ------------------------------------------------- |:-----------------:|
+| AUTHOR               | Author of the repository.                         | NONE * |
+| DOCPATH              | Path of the documentation.                        | NONE * (eg. `docs/`) |
+| DOCTHEME             | Name of the theme.                                | alabaster ([built-in themes](http://www.sphinx-doc.org/en/stable/theming.html#builtin-themes)) |
```
Comment (pri22296) [inline README.md]: apart from built-in themes, we also support any theme available in PyPI. Please mention that.

Comment (imujjwal96) [inline README.md]: Any list of all these themes, so I can reference them

Comment (sch00lb0y) [inline README.md]: check this https://pypi.python.org/pypi/sphinx-bootstrap-theme/

Comment (pri22296) [inline README.md]: No list as such. Specifying that any custom theme from pypi is supported should be enough.

Comment (imujjwal96) [inline README.md]: It is just a single theme. I was hoping if we could get a webpage containing a list of all these themes.

Comment (imujjwal96) [inline README.md]: okden

Comment (pri22296) [inline README.md]: does [this](https://pypi.python.org/pypi?%3Aaction=search&term=sphinx+theme&submit=search) help?

Comment (imujjwal96) [inline README.md]: Wouldn't be exclusive, would it?

Comment (imujjwal96) [inline README.md]: Any other changes, or these would be fine?

Comment (pri22296) [inline README.md]: No. I just shared it, if someone wanted to compile the list manually, but that would be overkill. Ignore it, No need to refer to a list.

Comment (pri22296) [inline README.md]: apart from these 2 changes, looks good to me.

**Pull request merged**